### ISSUE TITLE
Triggers enabled when using binary build with source image, when should be disabled.

### ISSUE THREAD

**Issue by GrahamDumpleton** [openshift/origin#15224]
When creating a binary build of type source strategy, along with using the ``--source-image`` and ``--source-image-path`` options, the triggers referencing the S2I builder as well as that of the source image are set to ``true`` instead of ``false``. This results in any updates to the S2I image or source image triggering a new build. The problem is that being a binary build there is no way for the build to pull in the source code. The build still runs, but with no source code and so fails, or produces an image which when it gets deployed fails due to missing source code in the final application image.

##### Version

oc v3.6.0-rc.0+98b3d56
kubernetes v1.6.1+5115d708d7
features: Basic-Auth

Server https://127.0.0.1:8443
openshift v3.6.0-rc.0+98b3d56
kubernetes v1.6.1+5115d708d7

##### Steps To Reproduce

Import image ``openshiftkatacoda/python35-wheelhouse``:

```
oc import-image openshiftkatacoda/python35-wheelhouse --confirm
```

Use the ``python35-wheelhouse`` S2I builder to generate a new image which will be used as the source image.

```
oc new-build --image-stream python35-wheelhouse --code https://github.com/openshift-katacoda/blog-django-py --name blog-wheelhouse
```

Result of this is:

```
$ oc get builds
NAME                TYPE      FROM          STATUS     STARTED              DURATION
blog-wheelhouse-1   Source    Git@832a277   Complete   About a minute ago   46s
```

Check out repository ``https://github.com/openshift-katacoda/blog-django-py``:

```
git clone https://github.com/openshift-katacoda/blog-django-py
```

Change directory into ``blog-django-py``.

Now create a binary build using ``python:3.5`` and the contents of the local directory, with reference to the source image created above.

```
oc new-build --binary --image-stream python:3.5 --source-image blog-wheelhouse --source-image-path /opt/app-root/cache/.:.s2i/cache/ --name blog
```

This should only create the build. The bug is that although it creates the build, it first up also runs a build, even though the source code for the build hasn't been supplied yet using ``oc start-build --from-dir``.

Result of ``oc new-build --binary`` is instead:

```
$ oc get builds
NAME                TYPE      FROM          STATUS     STARTED          DURATION
blog-1              Source                  Complete   9 seconds ago    5s
blog-wheelhouse-1   Source    Git@832a277   Complete   10 minutes ago   46s
```

That build ``blog-1`` should never have occurred.

The reason is that the triggers are enabled for both the S2I builder and the source image on the build configuration.

```
$ oc set triggers bc/blog
NAME               TYPE     VALUE                   AUTO
buildconfigs/blog  config                           true
buildconfigs/blog  image    blog-wheelhouse:latest  true
buildconfigs/blog  image    openshift/python:3.5    true
buildconfigs/blog  webhook  tBShps3QEFj81AEtRjGi
buildconfigs/blog  github   abCU11oPOCeRBL8_dBsl
```

If you now deploy the image created the pod will fail to start up as there is no code in it.

```
oc new-app --image-stream blog
```

Result is:

```
$ oc get pods
NAME                      READY     STATUS             RESTARTS   AGE
blog-1-build              0/1       Completed          0          1m
blog-1-deploy             1/1       Running            0          35s
blog-1-sbkt4              0/1       CrashLoopBackOff   2          34s
blog-wheelhouse-1-build   0/1       Completed          0          12m
```

Now run an actual build with the source code:

```
oc start-build blog --from-dir .
```

Now the application image was built with source code it runs.

```
$ oc get pods
NAME                      READY     STATUS      RESTARTS   AGE
blog-1-build              0/1       Completed   0          3m
blog-2-build              0/1       Completed   0          1m
blog-2-vqk8v              1/1       Running     0          8s
blog-wheelhouse-1-build   0/1       Completed   0          13m
```

Now trigger a rebuild of the source image.

```
oc start-build blog-wheelhouse
```

When the source image build completes, it triggers again another build of the application without source code and we get back to an application image that fails to deploy as source code is missing in it.

```
$ oc get builds
NAME                TYPE      FROM             STATUS     STARTED              DURATION
blog-1              Source                     Complete   5 minutes ago        5s
blog-2              Source    Binary@832a277   Complete   3 minutes ago        55s
blog-3              Source                     Complete   25 seconds ago       5s
blog-wheelhouse-1   Source    Git@832a277      Complete   16 minutes ago       46s
blog-wheelhouse-2   Source    Git@832a277      Complete   About a minute ago   49s

$ oc get pods
NAME                      READY     STATUS             RESTARTS   AGE
blog-1-build              0/1       Completed          0          5m
blog-2-build              0/1       Completed          0          3m
blog-2-vqk8v              1/1       Running            0          2m
blog-3-5zzqr              0/1       CrashLoopBackOff   2          35s
blog-3-build              0/1       Completed          0          41s
blog-3-deploy             1/1       Running            0          37s
blog-wheelhouse-1-build   0/1       Completed          0          16m
blog-wheelhouse-2-build   0/1       Completed          0          1m
```

##### Current Result

Triggers are being enabled on S2I image and source image used in a binary build. This causes rebuilds with no source code when those images change.

##### Expected Result

Triggers on a binary build using S2I image and source image should not be enabled.

##### Additional Information

The issue with the triggers being enabled only occurs when a source image is used with a binary build, specified using ``--source-image`` options. If it is a normal binary build without a source image, then triggers are disabled as expected.

**Comment by GrahamDumpleton** [comment]
Does the fact that the fix was merged mean that this will still make it into 3.6.0, or has it already missed out given rc.0 has already been created?

**Comment by bparees** [comment]
it was merged into master which now represents 3.7, so yeah, won't be in 3.6.  Wasn't deemed significant enough to put it in given where we are at in the 3.6 release cycle.
